llumnix/global_scheduler/migration_scheduler.py

```diff
@@ -21,12 +22,27 @@
 
 logger = init_logger(__name__)
 
+class PairMigrationConstraints(str, Enum):
+    """Target of Migration."""
+    NO_CONSTRAINTS = "NO_CONSTRAINTS"
```

Comment:
Given that you already have `InstanceType`, then it seems that you don't have to define string enums for the `PairMigrationConstraints`. This constraint could be a data structure (rather than a string) defining src and dst. For example, the `NO_CONSTRAINTS` could be translated to a src of the `NO_CONSTRAINTS` type to a dst of also the `NO_CONSTRAINTS` type. Similar for others.

Reply:
In this sense, each pair migration constraint means migrating from a certain instance type to another instance type

Reply:
> Given that you already have `InstanceType`, then it seems that you don't have to define string enums for the `PairMigrationConstraints`. This constraint could be a data structure (rather than a string) defining src and dst. For example, the `NO_CONSTRAINTS` could be translated to a src of the `NO_CONSTRAINTS` type to a dst of also the `NO_CONSTRAINTS` type. Similar for others.

 `PairMigrationConstraints` are primarily used by the global scheduler to provide the current migration decision. Within the migration scheduler, an additional step is now taken to translate PairMigrationConstraints into the src and dst of different instance type.